mitodl/ocw-studio
# Import ocw-to-hugo output into database

#### What are the relevant tickets?

Closes #8

#### What's this PR do?

Adds a management command, plus related celery tasks, to import ocw-to-hugo output from S3 into database models.

I added some s3 & celery related helper functions into `main/utils.py` on the assumption that they may be useful for something other than ocw2hugo import in the future.

There is a [code block](https://github.com/mitodl/ocw-studio/pull/10/files#diff-99dc8a77ec10391867c362180e6db632fe1e43cfcab696949ad1de969bff5540R60-R76) in this PR that deals with missing uids in some pages, this can be removed after https://github.com/mitodl/ocw-to-hugo/pull/155 merges and the S3 bucket is updated with the latest output.

#### How should this be manually tested?

Ask me for AWS keys and the S3 bucket name. Then run the following:
```
docker-compose run web python manage.py backpopulate_ocw_courses -b <bucket_name>
```
It should complete after about 45 minutes.
